lib/libyui_client/widgets/bargraph.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module LibyuiClient
+  module Widgets
+    # Class representing a DateField in the UI, namely YDateField.
+    class BarGraph < Widgets::Base
+      # Returns the list of segments in the bar graph. Each segment is defined by
+      # label and value. Segment length is defined by proportional ratio of the
+      # segment value. For instance, if segment values are 100, 200, 700: it means
+      # first segment will take 100/1000=10%, second 20% and third one 70% accordingly.
+      # @return [Array] array of [Hash] objects with :label and :value keys which
+      #                 store label and segment value for given segment
+      # @example Get segments from the bargraph with id "test_id"
+      # {
+      #   "class" : "YBarGraph",
+      #   "id" : "test_id",
+      #   "segments" :
+      #   [
+      #     {
+      #       "label" : "Windows\nused\n%1 MB",
+      #       "value" : 600
+      #     },
+      #     {
+      #       "label" : "Windows\nfree\n%1 MB",
+      #       "value" : 350
+      #     },
+      #     {
+      #       "label" : "Linux\n%1 MB",
+      #       "value" : 800
+      #     }
+      #   ],
+      # }
+      # @example
+      #   app.bargraph(id: 'test_id').segments
+      # @example
+      #   app.bargraph(id: 'test_id').segments[1][:value]
+      # @example
+      #   app.bargraph(id: 'test_id').segments[1][:label]
+      def segments
+        property(:segments)
+      end
+
+      # Returns the labels list of all segments in the bar graph.
+      # @return [Array] array of labels in order they are stored in segments hash.
+      # @example Get labels of the segments in the bargraph with id "test_id"
+      # @example
+      #   app.bargraph(id: 'test_id').labels
+      def labels
+        property(:segments).map { |segment| segment[:label] }
+      end
+
+      # Returns the values list of all segments in the bar graph.
+      # @return [Array] array of values in order they are stored in segments hash.
+      # @example Get values of the segments in the bargraph with id "test_id"
+      # @example
+      #   app.bargraph(id: 'test_id').values
+      def values
+        property(:segments).map { |segment| segment[:value] }
```

Comment:
Let's see the usecase for this, because calculating percentages will introduce rounding errors, etc. For the disk space, I bet we want to know GB value rather than percentage. So we can add another method to do that if we feel need in it.